### ISSUE TITLE
v1.4.0 doesn't launch

### ISSUE THREAD

**Issue by mindsolve** [xpipe-io/xpipe#67]
I've just updated from a 0.x release, but the application doesn't start anymore (no window is created).
When launching via the `xpiped_debug.bat` script, the below log is produced.
Uninstalling and reinstalling didn't change anything. The portable version doesn't work as well.

```
[mode] Initial setup
20:15:11:897 - info: Detected arguments {
    list=[]
}
20:15:11:907 - info: Detected system property user.dir=C:\Users\Felix\AppData\Local\XPipe\app\runtime\bin
20:15:11:907 - info: Detected xpipe property io.xpipe.app.logLevel=trace
20:15:11:907 - info: Detected xpipe property io.xpipe.app.sentryUrl=https://fd5f67ff10764b7e8a704bec9558c8fe@o1084459.ingest.sentry.io/6094279
20:15:11:907 - info: Detected xpipe property io.xpipe.app.arch=x86_64
20:15:11:907 - info: Detected xpipe property io.xpipe.app.fullVersion=true
20:15:11:907 - info: Detected xpipe property io.xpipe.beacon.debugOutput=true
20:15:11:907 - info: Detected xpipe property io.xpipe.app.writeLogs=true
20:15:11:907 - info: Detected xpipe property io.xpipe.app.developerMode=true
20:15:11:907 - info: Detected xpipe property io.xpipe.app.version=1.4.0
20:15:11:907 - info: Detected xpipe property io.xpipe.app.buildId=1bf6e69c-1834-3244-9990-250bf5aa3ce0
20:15:11:907 - info: Detected xpipe property io.xpipe.app.writeSysOut=true
20:15:11:907 - info: Detected xpipe property io.xpipe.app.build=1.4.0-202307211307
20:15:11:907 - info: Detected xpipe property io.xpipe.app.staging=false
20:15:11:907 - info: [mode] Finished initial setup
20:15:11:967 - error: Launcher command error occurred
java.lang.NullPointerException: Cannot invoke "io.xpipe.app.prefs.AppPrefs.startupBehaviour()" because the return value of "io.xpipe.app.prefs.AppPrefs.get()" is null
        at io.xpipe.app.launcher.LauncherCommand.getEffectiveMode(LauncherCommand.java:116)
        at io.xpipe.app.launcher.LauncherCommand.call(LauncherCommand.java:122)
        at io.xpipe.app.launcher.LauncherCommand.call(LauncherCommand.java:24)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at io.xpipe.app.launcher.LauncherCommand.runLauncher(LauncherCommand.java:64)
        at io.xpipe.app.core.mode.OperationMode.init(OperationMode.java:107)
        at io.xpipe.app.Main.main(Main.java:15)
```

**System info:**
- OS: Windows 10 x64 22H2

**Comment by crschnick** [comment]
Whoops, that seems to be my mistake on Windows.

**Comment by crschnick** [comment]
The fix for that should be released in approx 20 min when the CI pipeline is done.

**Comment by mindsolve** [comment]
Thanks for the quick fix! Good that is was spottet so quickly, or you might have had to answer a lot more issues ;)

**Comment by crschnick** [comment]
Yeah, that was unfortunate. This does not occur on Linux or my development environment on Windows, so it just slipped past that.

**Comment by crschnick** [comment]
I'm leaving this open though, as I need to investigate why there was no error handler pop up window.

**Comment by mindsolve** [comment]
Sure. The only (maybe) helpful thing I've observed there is that an error handler window opened when I downgraded to the previous release and then started the application. There a "update failed" (or similar) message appeared, with the option to report the problem either via Github (though with nothing prefilled), or directly.

**Comment by crschnick** [comment]
This is now fixed in https://github.com/xpipe-io/xpipe/releases/tag/1.4.1

I'm also keeping this open for other people that may have the same issue.

**Comment by crschnick** [comment]
I might also use this incident as a motivation to properly implement a beta program that you can enter where you can download releases early from within the application.

**Comment by joshoram80** [comment]
I didn't get a chance to test the 1.40 release which was broken but..

```
java.io.IOException: Input was closed before end was read
at io.xpipe.ext.procx.util.c.a(SourceFile:92)
at io.xpipe.ext.procx.l.exitAndWait(SourceFile:358)
at io.xpipe.ext.procx.l.prepareIntermediateTerminalOpen(SourceFile:102)
at io.xpipe.ext.procx.l.prepareTerminalOpen(SourceFile:89)
at io.xpipe.core.store.ShellStore.prepareLaunchCommand(ShellStore.java:27)
at io.xpipe.ext.base.action.LaunchAction$Action.execute(LaunchAction.java:29)
at io.xpipe.app.comp.storage.store.StoreEntryWrapper.executeDefaultAction(StoreEntryWrapper.java:197)
at io.xpipe.app.comp.storage.store.StoreEntryComp.lambda$createSimple$1(StoreEntryComp.java:81)
at io.xpipe.app.util.ThreadHelper.lambda$runFailableAsync$0(ThreadHelper.java:18)
at java.lang.Thread.run(Thread.java:1623)
Suppressed: java.io.IOException: The pipe is being closed
at java.io.FileOutputStream.writeBytes(Native Method)
at java.io.FileOutputStream.write(FileOutputStream.java:373)
at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:125)
at java.io.BufferedOutputStream.implFlush(BufferedOutputStream.java:252)
at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:240)
at io.xpipe.ext.proc.d.writeLine(SourceFile:42)
at io.xpipe.ext.procx.l.exitAndWait(SourceFile:356)
at io.xpipe.ext.procx.l.close(SourceFile:321)
at io.xpipe.ext.procx.l.prepareIntermediateTerminalOpen(SourceFile:94)
... 7 more
```

Whenever I try and launch a terminal

**Comment by crschnick** [comment]
That is now fixed in v1.4.2

**Comment by crschnick** [comment]
Closing this as hopefully everyone that had been affected has updated now to newer versions
